packages/core/src/codewhisperer/service/recommendationHandler.ts

```diff
@@ -158,8 +159,7 @@ export class RecommendationHandler {
         autoTriggerType?: CodewhispererAutomatedTriggerType,
         pagination: boolean = true,
         page: number = 0,
-        isSM: boolean = isSageMaker(),
-        retry: boolean = false
+        generate: boolean = isIamConnection(AuthUtil.instance.conn)
```

Comment:
This should no longer be a single function for both IAM and bearer token code path. The number of if-else is too much to comprehend. Not a blocker for this PR

Reply:
Please do lots of regression testing post merge

Reply:
Test /dev and /transform